crates/rpc/rpc-builder/src/constants.rs

```diff
@@ -20,7 +22,8 @@ pub const DEFAULT_MAX_BLOCKS_PER_FILTER: u64 = 100_000;
 pub const DEFAULT_MAX_LOGS_PER_RESPONSE: usize = 20_000;
 
 /// The default maximum number of concurrently executed tracing calls
-pub const DEFAULT_MAX_TRACING_REQUESTS: u32 = 25;
+pub static DEFAULT_MAX_TRACING_REQUESTS: Lazy<u32> =
+    Lazy::new(|| (max(std::thread::available_parallelism().unwrap().into(), 4) - 2) as u32);
```

Comment:
we can use saturatin_sub here and I don't want to use the unwrap here instead should default to the previous 25 if this call fails because this can undercount on certain systems

Reply:
Good suggestion. Done